qiskit_machine_learning/kernels/algorithms/quantum_kernel_trainer.py

```diff
@@ -0,0 +1,231 @@
+        return self._optimizer
+
+    @optimizer.setter
+    def optimizer(self, optimizer: Optimizer) -> None:
+        """Set the optimizer."""
+        self._optimizer = optimizer
+
+    @property
+    def initial_point(self) -> Optional[Sequence[float]]:
+        """Return initial point"""
+        return self._initial_point
+
+    @initial_point.setter
+    def initial_point(self, initial_point: Optional[Sequence[float]]) -> None:
+        """Set the initial point"""
+        self._initial_point = initial_point
+
+    def fit_kernel(
+        self,
+        data: np.ndarray,
+        labels: np.ndarray,
+    ) -> QuantumKernelTrainerResult:
+        """
+        Train the QuantumKernel by minimizing loss over the kernel parameters. The input
+        quantum kernel will not be altered, and an optimized quantum kernel will be returned.
+
+        Args:
+            data (numpy.ndarray): ``(N, D)`` array of training data, where ``N`` is the
+                              number of samples and ``D`` is the feature dimension
+            labels (numpy.ndarray): ``(N, 1)`` array of target values for the training samples
+
+        Returns:
+            QuantumKernelTrainerResult: the results of kernel training
+
+        Raises:
+            ValueError: No trainable user parameters specified in quantum kernel
+        """
+        # Number of parameters to tune
+        num_params = len(self._quantum_kernel.user_parameters)
+        if num_params == 0:
+            msg = "Quantum kernel cannot be fit because there are no user parameters specified."
+            raise ValueError(msg)
+
+        # Bind inputs to objective function
+        output_kernel = copy.deepcopy(self._quantum_kernel)
+
+        # Randomly initialize the initial point if one was not passed
+        if self._initial_point is None:
+            self._initial_point = algorithm_globals.random.random(num_params)
+
+        # Perform kernel optimization
+        loss_function = partial(
+            self._loss.evaluate, quantum_kernel=self.quantum_kernel, data=data, labels=labels
+        )
+        opt_results = self._optimizer.minimize(
+            fun=loss_function,
+            x0=self._initial_point,
+        )
+
+        # Return kernel training results
+        result = QuantumKernelTrainerResult()
+        result.optimizer_evals = opt_results.nfev
+        result.optimal_value = opt_results.fun
+        result.optimal_point = opt_results.x
+        result.optimal_parameters = dict(zip(output_kernel.user_parameters, opt_results.x))
+
+        # Return the QuantumKernel in optimized state
+        output_kernel.assign_user_parameters(result.optimal_parameters)
+        result.quantum_kernel = output_kernel
+
+        return result
+
+    def _str_to_loss(self, loss_str: str) -> KernelLoss:
+        if loss_str == "svc_loss":
+            loss_obj = SVCLoss()
+        else:
+            raise ValueError(f"Unknown loss {loss_str}!")
```

Comment:
This check is in the constructor now. Is this method now required?

Reply:
Good catch. I think I'd rather leave it here and remove from constructor